pilot/pkg/model/sidecar.go

```diff
@@ -447,20 +447,12 @@ func (sc *SidecarScope) GetEgressListenerForRDS(port int, bind string) *IstioEgr
 
 // Services returns the list of services imported by this egress listener
 func (ilw *IstioEgressListenerWrapper) Services() []*Service {
-	if ilw == nil {
```

Comment:
Why donot keep this to prevent developer misuse?

Reply:
I checked the contexts of IstioEgressListenerWrapper::Services(). There will be an immediate access of the other members of `IstioEgressListenerWrapper`. 
The production code is not ready for a nil value IstioEgressListenerWrapper, we can add it back if nil value does help in some flow.